test-app/tests/integration/components/bs-collapse-test.ts

```diff
@@ -44,18 +51,18 @@ module('Integration | Component | bs-collapse', function (hooks) {
       .hasClass(visibilityClass(), 'collapse has visibility class');
   });
 
-  test('setting collapse to false expands this item', async function (assert) {
-    let showAction = sinon.spy();
-    let shownAction = sinon.spy();
-    this.actions.show = showAction;
-    this.actions.shown = shownAction;
+  test('setting collapse to false expands this item', async function (this: Context, assert) {
+    const showAction = sinon.spy();
+    const shownAction = sinon.spy();
+    this.show = showAction;
+    this.shown = shownAction;
 
     this.set('collapsed', true);
-    await render(
+    await render<Context>(
```

Comment:
Something a colleague of mine came up with is to create a helper function to render the elements, while providing an object with the arguments. Would it make sense to introduce something like that here as well?

It usually looks something like this

```ts
renderWithArgs(args: BsCollapseSignature['Args']) {
  setAttributes(args) // I think this just loops over the properties of `args` and sets them using `this.set`.
  
  return render<BsCollapseSignature['Args']>(hbs`<BsCollapse
      {{! All arguments }}
    </BsCollapse>`);
}
```

This can then be used for all tests in the same class:

```ts
await renderWithArgs({
  show: false,
  shown: () => {},
  // etc...
});
```

This way the boilerplate of Glint context setting is handled by the helper function, and you don't need to bother with it in every `test` method.
Could something like this be interesting for `ember-bootstrap` as well?

Of course, if we decide to go for `gts`, the render context will just be the same as the function scope, but that might still be some time away.

Reply:
Interesting proposal!

Let's discuss it in #2093. I think we should land that one first. Will update the patterns here accordingly.

Reply:
But you would also need to render arbitrary things specific to the test, not just the component invocation with some args, like block content, named blocks, yielded stuff etc., and you can't do that with a template that's inside that helper? Or do I misunderstand?

Reply:
> But you would also need to render arbitrary things specific to the test, not just the component invocation with some args, like block content, named blocks, yielded stuff etc., and you can't do that with a template that's inside that helper? Or do I misunderstand?

Nope, that is one of the downsides. For components without block data it works like a dream...
I did forget that most components here do have yielded blocks, so it may not as useful as I initially thought 😅